Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h

```diff
@@ -185,6 +188,28 @@ namespace AWSCore
         {
         }
 
+        AwsApiRequestJob(bool queueOnSuccess,
+                         OnSuccessFunction onSuccess,
+                         bool queueOnFailure,
+                         OnFailureFunction onFailure,
+                         bool queueDelete,
+                         IConfig* config = GetDefaultConfig()
+        ) : AwsApiClientJobType(false, config)
+          , m_queueOnSuccess{ queueOnSuccess }
+          , m_onSuccess{ onSuccess }
+          , m_queueOnFailure{ queueOnFailure }
+          , m_onFailure{ onFailure }
+          , m_queueDelete{ queueDelete }
+        {
+        }
+
+        AwsApiRequestJob(OnSuccessFunction onSuccess,
+                         OnFailureFunction onFailure,
+                         IConfig* config = GetDefaultConfig()
+        ) : AwsApiRequestJob(true, onSuccess, true, onFailure, true, config)
+        {
+        }
```

Comment:
Minor: please make these two as private constructor?

I am ok with this approach, just a nitpick as there are extra queue success, failure and delete check in base class, can you create a GHI to us, so we should simplify/optimize this code?

Reply:
This was merged in, but i can do that in a follow-up PR